### PR TITLE
Fix drain stuck forever for unreplicated volume

### PR DESCRIPTION
**Pull request by cclhsu**
#### Proposed Changes ####

Fix drain stuck forever for unreplicated volume to resolve issue '[BUG] kubectl drain node gets stuck forever'

#### Linked Issues ####

Refs: # https://github.com/longhorn/longhorn/issues/2673

#### Further Comments ####

None

Signed-off-by: Clark Hsu <clark.hsu@suse.com>

### CODE REVIEW

File: controller/instance_manager_controller.go [hunk 1]
```diff
@@ -590,6 +590,10 @@ func (imc *InstanceManagerController) canDeleteInstanceManagerPDB(im *longhorn.I
 				hasHealthyReplicaOnAnotherNode = true
 				break
 			}
+			if r.Spec.HealthyAt == "" && r.Spec.FailedAt == "" && r.Spec.NodeID == imc.controllerID {
+				hasHealthyReplicaOnAnotherNode = true
```
Comment (shuo-wu) [inline controller/instance_manager_controller.go:594]: Please use an separate bool variable and add some comments for this case. e.g., `isUnusedReplicaOnCurrentNode`

**Review by shuo-wu — CHANGES_REQUESTED** [review]

File: controller/instance_manager_controller.go (hunk 1 above)
Comment (cclhsu) [reply controller/instance_manager_controller.go]: changed

File: controller/instance_manager_controller.go [hunk 2]
```diff
@@ -585,14 +585,21 @@ func (imc *InstanceManagerController) canDeleteInstanceManagerPDB(im *longhorn.I
...
 		hasHealthyReplicaOnAnotherNode := false
+		isUnusedReplicaOnCurrentNode := false
 		for _, r := range replicas {
 			if r.Spec.HealthyAt != "" && r.Spec.FailedAt == "" && r.Spec.NodeID != imc.controllerID {
 				hasHealthyReplicaOnAnotherNode = true
 				break
 			}
+			// Fix drained node `pod's disruption budget` issue, when LHV is is created through Longhorn UI,
+			// but has not yet attached and replicated.
+			if r.Spec.HealthyAt == "" && r.Spec.FailedAt == "" && r.Spec.NodeID == imc.controllerID {
```
Comment (shuo-wu) [inline controller/instance_manager_controller.go:596]:
```
If a replica has never been started, there is no data stored in this replica, and retaining it makes no sense for HA.
Hence Longhorn doesn't need to block the PDB removal for the replica.
This case typically happens on a newly created volume that hasn't been attached to any node.
```

You can add the Github link if you prefer.

**Review by shuo-wu — CHANGES_REQUESTED** [review]

File: controller/instance_manager_controller.go (hunk 2 above)
Comment (cclhsu) [reply controller/instance_manager_controller.go:599]: changed

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**